hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsLease.java

```diff
@@ -156,6 +185,29 @@ public void onFailure(Throwable throwable) {
     });
   }
 
+  private void spawnLeaseRenewTimer(String path, Integer leaseDuration) {
+    timer = new Timer();
+    timer.schedule(new TimerTask() {
+      @Override
+      public void run() {
+        try {
+          leaseID.set(callRenewLeaseAPI(path, leaseID.get(), tracingContext));
+        } catch (AzureBlobFileSystemException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    }, leaseDuration / 2, leaseDuration / 2);
+  }
```

Comment:
Can you please point out the code where this timer is stopped ? release lease code in AbfsBlobLease doesnt have any handling for timer.

Reply:
https://github.com/saxenapranav/hadoop/blob/0d316c4d12ada60ca055504735681a86e3ba692e/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsLease.java#L225-L227

The public function that can be called  is `free()`. This method will cancel the timer.

Reply:
Though there was a miss.
The timer was getting cancelled but not because of free method(it was not called from the rename block). It was getting cancel because when the blob was deleted and timerTask tried to renew it, it was getting 404 which was resulting in the break of timerTask.

Now, this has been rectified. Before deleting the blob, we are now cancelling the timer.